packages/frontend/src/ee/features/aiCopilot/components/ChatElements/AgentToolCalls/AgentToolCalls.tsx

```diff
@@ -0,0 +1,63 @@
+import { Box, List, Text, ThemeIcon } from '@mantine-8/core';
+import { IconTool } from '@tabler/icons-react';
+import { useParams } from 'react-router';
+import { useAiAgentThreadStreamToolCalls } from '../../../streaming/useAiAgentThreadStreamQuery';
+
+// TODO :: should be based on schemas
+const TOOL_DISPLAY_MESSAGES = {
+    findFields: 'Finding relevant fields',
+    generateBarVizConfig: 'Generating a bar chart',
+    generateCsv: 'Generating CSV file',
+    generateQueryFilters: 'Applying filters to the query',
+    generateTimeSeriesVizConfig: 'Generating a line chart',
+} as const;
+
+const TOOL_NAMES = Object.keys(TOOL_DISPLAY_MESSAGES);
+
+const AgentToolCalls = () => {
+    const { threadUuid } = useParams();
+    const toolCalls = useAiAgentThreadStreamToolCalls(threadUuid!);
+    if (toolCalls?.length === 0) return null;
+
+    return (
+        <Box bg="gray.0" p="xs" mb="sm">
+            <List
+                spacing="xs"
+                size="xs"
+                icon={
+                    <ThemeIcon
+                        size={14}
+                        radius="xl"
+                        variant="light"
+                        color="gray"
+                    >
+                        <IconTool size={8} />
```

Comment:
does `<MantineIcon` work with mantine 8?

Reply:
It does appear so... Yes?

Reply:
It does!